packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -0,0 +1,91 @@
+import fetch, { RequestInit } from 'node-fetch';
+
+type RequestMethods = 'post' | 'get';
+type RequestParams = {
+	method: 'post' | 'get';
+	headers: GenericDict;
+	body?: GenericDict;
+};
+type GenericDict = { [ key: string ]: string | number };
```

Comment:
Building on my comment below about changing the abstraction level, one nice perk is you wouldn't need these very dynamic types.

Each function could just parameters for exactly what info it needs, and an interface for the response data that is returns.